### ISSUE TITLE
RS3 graphics settings not persisting between client restarts

### ISSUE THREAD

**Issue by ajaypillay** [Adamcake/Bolt#83]
For whatever reason, I couldn't launch the RS3 and RuneLite clients. No error messages were being shown either. Clicking on "Play" didn't result in any window being opened. Couldn't get to the bottom of it, so I reinstalled Bolt via Flatpak.

However, now my RS3 graphics settings aren't persisting between launches of the client. I don't recall ever manually saving this anywhere before this. Any advice on how to further troubleshoot/fix this?

**Comment by Adamcake** [comment]
Before flatpak did you install from AUR? If so the issue was probably
missing optdepends?

Not saving graphics settings is strange though, the client should save all
that in `.var/app/com.adamcake.Bolt/data/bolt-launcher/Jagex`

**Comment by ajaypillay** [comment]
No, I'm on Fedora 40 and I previously installed it via Flatpak too.

Do you know which specific file it's located in or have an example of what that should look like?

I checked the contents of:

- `.var/app/com.adamcake.Bolt/data/bolt-launcher/Jagex/launcher/preferences.cfg`
- `.var/app/com.adamcake.Bolt/data/bolt-launcher/Jagex/RuneScape/GlobalSettings.jcache`
- `.var/app/com.adamcake.Bolt/data/bolt-launcher/Jagex/RuneScape/Settings.jcache`

But none of them contained anything that hints at storing graphics settings. Tried looking for a log file too to see if there are any hidden errors being thrown but couldn't find such a file.

**Comment by Adamcake** [comment]
Seems to be Settings.jcache, the third one you mentioned. It's stored in some SQL-based binary format but I can see the names of a few graphics settings in it:
![image](https://github.com/user-attachments/assets/b2a321ad-8040-421b-89f6-0b7fd83995f6)

**Comment by ajaypillay** [comment]
It's an SQLite file. Here's the output of `file Settings.jcache`:
```
Settings.jcache: SQLite 3.x database, application id 1701737810, user version 2, last written using SQLite version 3025001, file counter 2681, database pages 7, cookie 0x3, schema 4, UTF-8, version-valid-for 2681
```
I used https://sqliteviewer.app/ to view the file.

![image](https://github.com/user-attachments/assets/ff2ae3da-926d-4c6d-9374-2baf621ba79d)

And that's the only data I have saved in the `Config` table in that file. There are two other tables, `player` and `vt-varc`. `player` just has a `uid` which I'm assuming is related to the character, and it isn't entirely clear to me what `vt-varc` is, perhaps some kind of virtual table.

I opened the file using https://sqlitebrowser.org/ and manually tried inserting a setting based on your screenshot:
```
INSERT INTO Config (KEY, DATA) VALUES ("DrawDistance_MaxLower", 3);
```
I committed it and ensured it was saved (the RS3 client was open while doing this). I then restarted the client without closing Bolt and it persisted my settings. The table is now populated correctly, here's the first few rows:
![image](https://github.com/user-attachments/assets/8e8c9534-7b73-4df1-9c09-24f854f7c209)

I closed the RS3 client, relaunched Bolt, and the settings now correctly persist. Not too sure what this triggered, maybe some read/write permissions got modified when I interacted with the file through the DB browser which cascaded to the client being able to interact with it correctly? In any case, looks like this is solved for now. Thanks for the help and providing that screenshot.
